module-apps/application-meditation/widgets/IntervalBox.cpp

```diff
@@ -73,6 +74,30 @@ std::chrono::seconds IntervalBox::getIntervalValue() noexcept
     }
 }
 
+void IntervalBox::setIntervalValue(std::chrono::minutes newValue)
+{
+    const auto it = std::find(std::begin(chimeIntervals), std::end(chimeIntervals), newValue);
+    if (it == std::end(chimeIntervals)) {
+        LOG_ERROR("Incorrect parameter value! The default value is set.");
+        newValue = defaultChimeIntervals;
+    }
+
+    if (auto setTime = timerSetter->getTime(); newValue < setTime) {
+        chimeSpinner->setCurrentValue(intervalToString(newValue));
+    }
+}
+
+void IntervalBox::setOnChangeCallback(std::function<void(std::chrono::minutes newValue)> &callback)
+{
+    if (chimeSpinner != nullptr) {
+        chimeSpinner->setOnValueChangeCallback([&](const UTF8 &newValue) {
+            if (callback != nullptr) {
```

Comment:
I hope `callback` will be still a valid object by the time `chimeSpinner` invoke it ;)

Reply:
this spinner is used only by the main meditation window and there is an instance of parameters, i.e. also a callback, so there is no way that this callback does not already exist at the time of invocation in this spinner